src/Components/Web.Extensions/src/HeadManagement/HeadManagementRenderTreeBuilderExtensions.cs

```diff
@@ -0,0 +1,13 @@
+using System.Text.Json;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    internal static class HeadManagementRenderTreeBuilderExtensions
+    {
+        public static void BuildHeadElementComment<TElement>(this RenderTreeBuilder builder, int sequence, TElement element)
+        {
+            builder.AddMarkupContent(sequence, $"<!--Head:{JsonSerializer.Serialize(element, JsonSerializerOptionsProvider.Options)}-->");
```

Comment:
Is there any risk that the JSON data might contain the character sequence `-->` for unrelated reasons?

Reply:
It looks like `JsonSerializer.Serialize` encodes `>` and `<`, so I believe this should be safe (I've tested it, and including `-->` as a property value in the instance being serialized didn't cause any issues).